### ISSUE TITLE
Cannot allocate 18446744071562067968 bytes on erlang:binary_to_term/1

### ISSUE THREAD

**Issue by josevalim** [erlang/otp#6393]
**Describe the bug**

Take this code:

```erlang
Bin = binary:copy(<<0>>, 2 * 1024 * 1024 * 1024),
erlang:binary_to_term(erlang:term_to_binary(Bin)).
```

It fails with reason:

```
binary_alloc: Cannot allocate 18446744071562067968 bytes of memory (of type "binary").

Crash dump is being written to: erl_crash.dump...done
```

**Expected behavior**

I would expect it to succeed or a clearer error message. :)

**Affected versions**

I have seen the failure on Erlang/OTP 24 and 25, on both Linux and macOS.

**Comment by max-au** [comment]
Good old integer overflow... size appears to be negative.

**Comment by sverker** [comment]
Fixed by #6401 merged to maint for OTP 25.2.
